Mathlib/AlgebraicGeometry/EllipticCurve/DivisionPolynomial/Basic.lean

```diff
@@ -0,0 +1,463 @@
+import Mathlib.NumberTheory.EllipticDivisibilitySequence
+
+/-!
+# Division polynomials of Weierstrass curves
+
+This file defines certain polynomials associated to division polynomials of Weierstrass curves.
+These are defined in terms of the auxiliary sequences of normalised elliptic divisibility sequences
+defined in `Mathlib.NumberTheory.EllipticDivisibilitySequence`.
+
+## Mathematical background
+
+Let $W$ be a Weierstrass curve over a commutative ring $R$. The sequence of $n$-division polynomials
+$\psi_n \in R[X, Y]$ of $W$ is the normalised elliptic divisibility sequence with initial values
+ * $\psi_0 := 0$,
+ * $\psi_1 := 1$,
+ * $\psi_2 := 2Y + a_1X + a_3$,
+ * $\psi_3 := 3X^4 + b_2X^3 + 3b_4X ^ 2 + 3b_6X + b_8$, and
+ * $\psi_4 := \psi_2 \cdot
+    (2X^6 + b_2X^5 + 5b_4X^4 + 10b_6X^3 + 10b_8X^2 + (b_2b_8 - b_4b_6)X + (b_4b_8 - b_6^2))$.
+
+Furthermore, define the associated sequences $\phi_n, \omega_n \in R[X, Y]$ by
+ * $\phi_n := X\psi_n^2 - \psi_{n + 1}\psi_{n - 1}$, and
+ * $\omega_n := \tfrac{1}{2} \cdot (\psi_{2n} / \psi_n - \psi_n(a_1\phi_n + a_3\psi_n^2))$.
+
+Note that $\omega_n$ is always well-defined as a polynomial. As a start, it can be shown by
+induction that $\psi_n$ always divides $\psi_{2n}$, so that $\psi_{2n} / \psi_n$ is always
+well-defined as a polynomial, while division by 2 is well-defined when $R$ has characteristic
+different from 2. In general, it can be shown that 2 always divides the polynomial
+$\psi_{2n} / \psi_n - \psi_n(a_1\phi_n + a_3\psi_n^2)$ in the characteristic zero universal ring
+$\mathbb{Z}[A_i][X, Y]$ of $W$, where $A_i$ are indeterminates. Then $\omega_n$ can be equivalently
+defined as its image under the associated universal morphism $\mathbb{Z}[A_i][X, Y] \to R[X, Y]$.
+
+Now, in the coordinate ring $R[W]$, note that $\psi_2^2$ is congruent to the polynomial
+$\Psi_2^{[2]} := 4X^3 + b_2X^2 + 2b_4X + b_6 \in R[X]$. As such, in $R[W]$, the recurrences
+associated to a normalised elliptic divisibility sequence show that $\psi_n / \psi_2$ are congruent
+to certain polynomials in $R[X]$. In particular, define $\tilde{\Psi}_n \in R[X]$ as the auxiliary
+sequence for a normalised elliptic divisibility sequence with initial values
+ * $\tilde{\Psi}_0 := 0$,
+ * $\tilde{\Psi}_1 := 1$,
+ * $\tilde{\Psi}_2 := 1$,
+ * $\tilde{\Psi}_3 := \psi_3$, and
+ * $\tilde{\Psi}_4 := \psi_4 / \psi_2$.
+
+The corresponding normalised elliptic divisibility sequence $\Psi_n \in R[X, Y]$ is then given by
+ * $\Psi_n := \tilde{\Psi}_n\psi_2$ if $n$ is even, and
+ * $\Psi_n := \tilde{\Psi}_n$ if $n$ is odd.
+
+Furthermore, define the associated sequences $\Psi_n^{[2]}, \Phi_n \in R[X]$ by
+ * $\Psi_n^{[2]} := \tilde{\Psi}_n^2\Psi_2^{[2]}$ if $n$ is even,
+ * $\Psi_n^{[2]} := \tilde{\Psi}_n^2$ if $n$ is odd,
+ * $\Phi_n := X\Psi_n^{[2]} - \tilde{\Psi}_{n + 1}\tilde{\Psi}_{n - 1}$ if $n$ is even, and
+ * $\Phi_n := X\Psi_n^{[2]} - \tilde{\Psi}_{n + 1}\tilde{\Psi}_{n - 1}\Psi_2^{[2]}$ if $n$ is odd.
```

Comment:
These two lines are broken in the [documentation](https://leanprover-community.github.io/mathlib4_docs/Mathlib/AlgebraicGeometry/EllipticCurve/DivisionPolynomial/Basic.html) and I'm not sure why.

Reply:
Seems like it's an issue with `_` being both part of markdown and LaTeX, see e.g. [here](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Contribution.20to.20doc-gen4.3F/near/443257844).